core/contract.go

```diff
@@ -126,11 +124,40 @@ func (c *Contract) ClassHash() (*felt.Felt, error) {
 
 // Root returns the root of the contract storage.
 func (c *Contract) Root() (*felt.Felt, error) {
-	if cStorage, err := storage(c.Address, c.txn); err != nil {
+	cStorage, err := storage(c.Address, c.txn)
+	if err != nil {
+		return nil, err
+	}
+	return cStorage.Root()
+}
+
+// Commitment returns contract storage commitment
+func (c *Contract) Commitment() (*felt.Felt, error) {
```

Comment:
I still think commitment is a state thing. Contract has no use for commitment, but state does.

Reply:
reverted